src/drivers/gitlab.js

```diff
@@ -19,39 +19,48 @@ class Gitlab {
 
     this.token = token;
     this.repo = repo;
+  }
+
+  async project_path() {
+    const api_path = await this.repo_base();
+    const project_path = encodeURIComponent(
+      this.repo.replace(api_path, '').substr(1)
```

Comment:
I'd rather explicitly remove the first occurence instead of running `replace`, as it could have unintended consequences in some edge cases.

Reply:
replace only removes the first occurence unless u use regex /g

Reply:
@0x2b3bfa0

Reply:
Yes, but it doesn't neccessarily have to be the beginning of the string. Maybe it's just paranoid, but we could do something like this (demi-serious pseudocode):

```javascript
if (!this.repo.startsWith(api_path)) throw new Error("unbelievable error");
this.repo.substr(api_path.length + 1)
```

Reply:
I dont follow you. Replace over substr?

Reply:
Well, I actually don't care of which one, but I thought that it would be nice to check first that the string actually starts with the substring we're going to remove.

Reply:
I dont think that defensive programming is not needed here? If that happens is because the repo is not valid.
If repro_base is working (it is) is always returning a subtr of repo. I cant think right now a case where that happens

Reply:
You're right! 🤦  The input comes from a controlled source, and there isn't any obvious case where this could pose any issue. Sorry for the false positive.